### PR TITLE
[VAULT-34216] UI: Namespace picker feature branch

### PR DESCRIPTION
**Pull request by beagins**
| 🥁 Drumroll: merging to `main` ❗ |
| -- |

### Description
- [x] Updates to the Namespace Picker:
  - Implement using `Hds::Dropdown` component
  - Support search across namespaces
  - Support navigate to namespace on Enter
  - Action buttons to refresh & manage namespaces (if user has appropriate capabilities)
  - Implemented frontend infinite scroll to handle many namespaces
  - Update component from `js` to `ts`
  - Handle errors and edge cases
  - Test coverage
- [x] Enterprise tests are passing ✅
  > 248 tests completed in 161597 milliseconds, with 0 failed, 10 skipped, and 0 todo.
1468 assertions of 1468 passed, 0 failed.

Screenshot 2025-05-01 at 3 30 24 PM
Screenshot 2025-05-01 at 3 30 38 PM
Screenshot 2025-05-01 at 3 30 51 PM
Screenshot 2025-05-01 at 3 31 33 PM

### TODO only if you're a HashiCorp employee
- [ ] **Backport Labels:** If this fix needs to be backported, use the appropriate `backport/` label that matches the desired release branch. Note that in the CE repo, the latest release branch will look like `backport/x.x.x`, but older release branches will be `backport/ent/x.x.x+ent`.
    - [ ] **LTS**: If this fixes a critical security vulnerability or [severity 1](https://www.hashicorp.com/customer-success/enterprise-support) bug, it will also need to be backported to the current [LTS versions](https://developer.hashicorp.com/vault/docs/enterprise/lts#why-is-there-a-risk-to-updating-to-a-non-lts-vault-enterprise-version) of Vault. To ensure this, use **all** available enterprise labels.
- [ ] **ENT Breakage:** If this PR either 1) removes a public function OR 2) changes the signature
  of a public function, even if that change is in a CE file, _double check_ that
  applying the patch for this PR to the ENT repo and running tests doesn't
  break any tests. Sometimes ENT only tests rely on public functions in CE
  files.
- [ ] **Jira:** If this change has an associated Jira, it's referenced either
  in the PR description, commit message, or branch name.
- [ ] **RFC:** If this change has an associated RFC, please link it in the description.
- [ ] **ENT PR:** If this change has an associated ENT PR, please link it in the
  description. Also, make sure the changelog is in this PR, _not_ in your ENT PR.

### CODE REVIEW

File: ui/lib/core/addon/utils/keyboard-keys.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+ */
+export enum KEYS {
+  ENTER = 'Enter',
+  ESC = 'Escape',
+  TAB = 'Tab',
+  LEFT = 'ArrowLeft',
+  UP = 'ArrowUp',
+  RIGHT = 'ArrowRight',
+  DOWN = 'ArrowDown',
+  T = 'F5',
```
Comment (lane-wetmore) [inline ui/lib/core/addon/utils/keyboard-keys.ts:18]: Why does `T` map to `F5`?

**Review by lane-wetmore — DISMISSED** [review]

File: ui/lib/core/addon/utils/keyboard-keys.ts (hunk 1 above)
Comment (beagins) [reply ui/lib/core/addon/utils/keyboard-keys.ts:18]: That's a good question!! I was replacing these keys one-for-one from `key-codes.js`. In the deprecated `key-codes` file, the `T` key is mapped to key code `116`, which is `F5`. But I'm not sure why they used `T` instead of `F5` as the key. The deprecated `key-codes` file was created 7 years ago and hasn't really been changed much since then. It looks like there are currently no references to the `T` `key-code`... are there any strong opinions on: 
* removing the unused property 
* vs renaming the key from `T` to `F5` 
* vs leaving it as-is? 

My vote is removing the unused prop. Thanks for catching this Lane!

Comment (lane-wetmore) [reply ui/lib/core/addon/utils/keyboard-keys.ts]: Sounds good! I wasn't sure where these were all from, and that once seemed a little opaque, so good follow-up!

**Review by lane-wetmore — APPROVED** [review]

**Review by Monkeychip — APPROVED** [review]
